dashboard/views.py

```diff
@@ -150,3 +152,36 @@ def post(self, request):
         return Response(
             data=enrollment.json
         )
+
+class UserPaymentStatus(APIView):
+    """
+    returns list of program id and status weather user has paid for atleast one course in selected program.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def get(self, request):
+        data = []
+        for program_enrollment in ProgramEnrollment.objects.filter(
+                user=request.user
+        ).iterator():
+            mmtrack = get_mmtrack(
+                program_enrollment.user,
+                program_enrollment.program
+            )
+
+            has_paid_for_course = []
+            for edx_course_key in mmtrack.edx_course_keys:
+                has_paid_for_course.append(mmtrack.is_enrolled_mmtrack(edx_course_key))
+
+            data.append({
+                str(program_enrollment.program.id): any(has_paid_for_course)
+            })
```

Comment:
We can use `MMTrack.has_paid_for_any_in_program` to determine if the user has paid for any course in the program.

```python
mmtrack = get_mmtrack(
    program_enrollment.user,
    program_enrollment.program
)
data.append({
    str(program_enrollment.program.id): mmtrack.has_paid_for_any_in_program()
})
```